### PR TITLE
Making a minimal mix

### LINKED ISSUE #704: Create a minimal mix
Currently, the “os-core” bundle and its “filesystem” package is a hard requirement in Mixer. In order to enable a minimal mix for 3rd party, make the “filesystem” package optional. In case the “filesystem” package is found in any of the mix bundles, then Mixer should install the “filesystem” package first. If it’s not found in any of the mix bundles, then just don’t install it. 

The “os-core” bundle should still be a hard requirement in Mixer as it is required by swupd-client.
If the "os-release" file is not found, then Mixer should create and inject a default one.

### PR DESCRIPTION
**Pull request by ashleshaAtrey**
Currently, the “os-core” bundle and its “filesystem” package is a hard requirement in Mixer.
In order to enable a minimal mix for 3rd party, “filesystem” package is made optional.
In case the “filesystem” package is found in any of the mix bundles,
then Mixer should install the “filesystem” package first.
If it’s not found in any of the mix bundles, then filesystem is not
installed.

The “os-core” bundle is still a hard requirement in Mixer.
If the "os-release" file is not found, then Mixer creates and injects a default one.
fixes #704

Signed-off-by: Ashlesha Atrey <ashlesha.atrey@intel.com>

### CODE REVIEW

**Comment by reaganlo** [comment]
Instead of failing if the "os-release" file is not found, can Mixer create and inject a default one. I have updated the #704. Also please check if there are other hard requirements.

**Comment by ashleshaAtrey** [comment]
@reaganlo made the changes. Ready for review.

File: builder/bundles.go [hunk 1]
```diff
@@ -535,7 +532,7 @@ func buildOsCore(b *Builder, packagerCmd []string, chrootDir, version string) er
 		return err
 	}
 
-	if err := fixOSRelease(b, filepath.Join(chrootDir, "usr/lib/os-release"), version); err != nil {
+	if err := fixOSRelease(b, filepath.Join(chrootDir, "usr/lib/os-release"), version, chrootDir); err != nil {
```
Comment (reaganlo) [inline builder/bundles.go]: Its not part of this change, but can we call this function `updateOSReleaseFile()`?
Lets also add a print statement inside the function, `fmt.Println("... updating os-release file")`

File: builder/bundles.go [hunk 2]
```diff
@@ -1181,3 +1187,65 @@ func getClosestAncestorOwner(path string) (int, int, error) {
...
+	}
+	_, err = fmt.Fprintln(f, "ID="+"clear-linux-os")
+	if err != nil {
+		return err
+	}
+	_, err = fmt.Fprintln(f, "ID_LIKE="+"clear-linux-os")
+	if err != nil {
+		return err
+	}
+	_, err = fmt.Fprintln(f, "VERSION_ID="+"")
```
Comment (reaganlo) [inline builder/bundles.go]: In the case the os-release file is not found, its contents are created and written. Its then opened again to update the VERSION_ID. Instead can we just update the VERSION_ID here itself.
So maybe instead of calling `createOSReleaseFile()` inside `updateOSReleaseFile()`, we can call it inside `buildOsCore()`. Something like this:

```
	var filename string
	if b.Config.Mixer.OSReleasePath != "" {
		filename = b.Config.Mixer.OSReleasePath
	} else {
		filename = filepath.Join(chrootDir, "usr/lib/os-release")
	}

	if _, err := os.Stat(filename); err != nil {
		if os.IsNotExist(err) {
			if err = createOSReleaseFile(filename, version, chrootDir); err != nil {
				return errors.Wrap(err, "couldn't create os-release file")
			}
		} else {
			return err
		}
	} else {
		if err := updateOSReleaseFile(b, filename, version, chrootDir); err != nil {
			return errors.Wrap(err, "couldn't update os-release file")
		}
	}
```

File: builder/bundles.go [hunk 3]
```diff
@@ -1181,3 +1187,65 @@ func getClosestAncestorOwner(path string) (int, int, error) {
...
+	}
+	_, err = fmt.Fprintln(f, "PRETTY_NAME="+"\""+"Clear Linux OS"+"\"")
+	if err != nil {
+		return err
+	}
+	_, err = fmt.Fprintln(f, "ANSI_COLOR="+"\""+"1;35"+"\"")
+	if err != nil {
+		return err
+	}
+	_, err = fmt.Fprintln(f, "HOME_URL="+"\""+chrootDir+"\"")
```
Comment (reaganlo) [inline builder/bundles.go]: I think @bryteise mentioned the default value for "HOME_URL" to be the `Swupd.CONTENTURL` in the builder.conf

File: builder/bundles.go (hunk 2 above)
Comment (ashleshaAtrey) [reply builder/bundles.go]: @reaganlo updated the createOSReleaseFile to update version_ID as well.
if createOSReleaseFile is successful, it return from function.
Still calling  createOSReleaseFile from UpdateOSReleaseFile.

File: builder/bundles.go [hunk 4]
```diff
@@ -1123,14 +1122,19 @@ func createVersionsFile(baseDir string, packagerCmd []string) error {
...
 	var err error
 	var f *os.File
 	if b.Config.Mixer.OSReleasePath != "" {
 		f, err = os.Open(b.Config.Mixer.OSReleasePath)
 	} else {
+		_, err = os.Stat(filename)
+		if err != nil {
+			err = createOSReleaseFile(filename, homeURL, version)
+			return err
+		}
```
Comment (reaganlo) [inline builder/bundles.go]: Shouldn't we be checking for `os.IsNotExist(err)`
Something like:
```
		if _, err = os.Stat(filename); os.IsNotExist(err) {
			return createOSReleaseFile(filename, homeURL, version)
		}
		if err != nil {
			return err
		}
```

Comment (ashleshaAtrey) [reply builder/bundles.go]: yes, will fix it.

**Comment by reaganlo** [comment]
Updating git commit message and merging
